api/main.py

```diff
@@ -138,11 +139,12 @@ async def post_user(
                         detail=f"User group does not exist with name: \
 {group_name}")
                 group_obj.append(group)
-        obj = await db.create(User(
+        profile = await db.create(UserProfile(
                                 username=username,
                                 hashed_password=hashed_password,
                                 groups=group_obj
                                 ))
+        obj = await db.create(User(profile=profile))
```

Comment:
Looks like the profile is created twice, first as a separate document and then again as nested in User.

Reply:
A profile is created just once. The already created profile document will be assigned to `User.profile`.
Please see the below documents from my local setup:
UserProfile `jeny`:
```
{ "_id" : ObjectId("64abdcc7ed84334468b3acbe"), "username" : "jeny", "hashed_password" : "$2b$12$QhsvplXElm34.HiF2rF6yO3WlYMLm1ZyDyo2hdtp3vDv0fdTjouqK", "groups" : [ { "_id" : ObjectId("64abe12e40679ecb42473768"), "name" : "kernelci" } ] }
```
User:
```
{ "_id" : ObjectId("64abdcc7ed84334468b3acbf"), "active" : true, "profile" : { "_id" : ObjectId("64abdcc7ed84334468b3acbe"), "username" : "jeny", "hashed_password" : "$2b$12$QhsvplXElm34.HiF2rF6yO3WlYMLm1ZyDyo2hdtp3vDv0fdTjouqK", "groups" : [ { "_id" : ObjectId("64abe12e40679ecb42473768"), "name" : "kernelci" } ] } }

```

Reply:
Yes, the UserProfile data is therefore created twice as "username" etc. is embedded in User.

I think we can just create a User document, then you can search using the User.profile.username attribute.

Reply:
This is the same as we are embedding user groups models in `UserProfile` model. We are just assigning user group objects while creating user profiles.

If we directly create a `User` model then it won't have any `UserProfile` model we can map to in the database.
```
{ "_id" : ObjectId("64afabd100848e93a69920ad"), "active" : true, "profile" : { "_id" : null, "username" : "jeny", "hashed_password" : "$2b$12$.x886aNDVsV1kUPEP0kOJ.FO29QYlrK1KJVjb7klciI4UtX28BKvS", "groups" : [ ] } }
```

Reply:
Yes but for groups, it makes sense to have a separate one as they're used in several places.  Each profile will only be used by a single user.